libarchive/archive_read_support_format_tar.c

```diff
@@ -1323,28 +1338,60 @@ header_common(struct archive_read *a, struct tar *tar,
 		archive_entry_set_mtime(entry, tar_atol(header->mtime, sizeof(header->mtime)), 0);
 	}
 
-	/* Update size information as appropriate */
-	if (!archive_entry_size_is_set(entry)) {
-		tar->entry_bytes_remaining = tar_atol(header->size, sizeof(header->size));
-		if (tar->entry_bytes_remaining < 0) {
-			tar->entry_bytes_remaining = 0;
-			archive_set_error(&a->archive, ARCHIVE_ERRNO_MISC,
-					  "Tar entry has negative size");
-			return (ARCHIVE_FATAL);
-		}
-		if (tar->entry_bytes_remaining > entry_limit) {
-			tar->entry_bytes_remaining = 0;
-			archive_set_error(&a->archive, ARCHIVE_ERRNO_MISC,
-					  "Tar entry size overflow");
-			return (ARCHIVE_FATAL);
-		}
-		if (!tar->realsize_override) {
-			tar->realsize = tar->entry_bytes_remaining;
-		}
-		archive_entry_set_size(entry, tar->realsize);
-	} else if (tar->realsize_override) {
-		tar->entry_bytes_remaining = tar->realsize;
-		archive_entry_set_size(entry, tar->realsize);
+	/* Reconcile the size info. */
```

Comment:
At this point, we've read all the extended header information and are now considering the last pieces of the "common" header fields such as size, mode, uid, etc.

Since we now have seen all possible size information, we're now ready to use that to figure out the actual size on disk and the size in the archive.